dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/loadbalance/StatisticsLoadBalance.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.dubbo.rpc.cluster.loadbalance;
+
+import org.apache.dubbo.common.URL;
+import org.apache.dubbo.common.config.ConfigurationUtils;
+import org.apache.dubbo.rpc.cluster.loadbalance.statistics.CpuUsageService;
+import org.apache.dubbo.rpc.cluster.loadbalance.statistics.CpuUsageServiceImpl;
+import org.apache.dubbo.rpc.Invocation;
+import org.apache.dubbo.rpc.Invoker;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class StatisticsLoadBalance extends AbstractLoadBalance {
+
+    private Map<String, Float> cpuUsage = new HashMap<>();
+
+    public StatisticsLoadBalance() {
+        long timeToLive = Long.parseLong(ConfigurationUtils.getProperty("time.to.live"));
+        long collectCpuUsageInMill = Long.parseLong(ConfigurationUtils.getProperty("mill.to.collect.cpu.usage"));
+        CpuUsageService cpuUsageService = new CpuUsageServiceImpl(timeToLive, collectCpuUsageInMill);
```

Comment:
CpuUsageService should be created by dubbo as  a remote stub instead of 'new' directly.

Reply:
What do you think about using SPI mechanism? (http://dubbo.apache.org/en-us/blog/introduction-to-dubbo-spi.html)

Reply:
Hi, 

I did some local test, the proposed pull request could not work correctly. If you start two provider and a consumer, you can see that cpu usage is not being sent from provider to consumer.

I agree with @guohao that, we need to call the CpuUsageService remotely to add listener rather that `new` directly, and if there are multiple providers, we have to call the addListener method for each provider. 

The problem is the StatisticsLoadBalance is a bean managed by Spring, so it is impossible to get the provider ip addresses when initializing StatisticsLoadBalance class. So we have to delay the call the addListener to the `doSelect` method.

To obtain a proxy for remote service, you may checkout `org.apache.dubbo.monitor.dubbo.DubboMonitorFactory#createMonitor`, there are some example code.

Since it is lazy to obtain the cpu usage, the load balance may not working very well during the initial period of time (lack of cpu usage stat), so there should be some fall back strategies, once the cpu stat are collected, the stat based load balance can be activated.